src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -45,41 +47,27 @@ public class TrieStore : ITrieStore, IPruningTrieStore
     private bool _lastPersistedReachedReorgBoundary;
     private Task _pruningTask = Task.CompletedTask;
     private readonly CancellationTokenSource _pruningTaskCancellationTokenSource = new();
-
-    public TrieStore(IKeyValueStoreWithBatching? keyValueStore, ILogManager? logManager)
-        : this(keyValueStore, No.Pruning, Pruning.Persist.EveryBlock, logManager)
-    {
-    }
-
-    public TrieStore(INodeStorage nodeStorage, ILogManager? logManager)
-        : this(nodeStorage, No.Pruning, Pruning.Persist.EveryBlock, logManager)
-    {
-    }
-
-    public TrieStore(
-        IKeyValueStoreWithBatching? keyValueStore,
-        IPruningStrategy? pruningStrategy,
-        IPersistenceStrategy? persistenceStrategy,
-        ILogManager? logManager) : this(new NodeStorage(keyValueStore), pruningStrategy, persistenceStrategy, logManager)
-    {
-    }
+    private readonly IPruningConfig _pruningConfig;
 
     public TrieStore(
-        INodeStorage? nodeStorage,
-        IPruningStrategy? pruningStrategy,
-        IPersistenceStrategy? persistenceStrategy,
-        ILogManager? logManager)
+        INodeStorage nodeStorage,
+        IPruningStrategy pruningStrategy,
+        IPersistenceStrategy persistenceStrategy,
+        IPruningConfig pruningConfig,
+        ILogManager logManager)
     {
-        _logger = logManager?.GetClassLogger<TrieStore>() ?? throw new ArgumentNullException(nameof(logManager));
-        _nodeStorage = nodeStorage ?? throw new ArgumentNullException(nameof(nodeStorage));
-        _pruningStrategy = pruningStrategy ?? throw new ArgumentNullException(nameof(pruningStrategy));
-        _persistenceStrategy = persistenceStrategy ?? throw new ArgumentNullException(nameof(persistenceStrategy));
+        _logger = (ILogger)logManager?.GetClassLogger<TrieStore>();
+        _nodeStorage = nodeStorage;
+        _pruningStrategy = pruningStrategy;
+        _persistenceStrategy = persistenceStrategy;
         _publicStore = new TrieKeyValueStore(this);
         _persistedNodeRecorder = PersistedNodeRecorder;
+        _pruningConfig = pruningConfig;
+        _maxDepth = pruningConfig.PruningBoundary;
```

Comment:
You capture `_maxDepth` to a field but not other parameters, why the inconsistency?